java/pattern/variable/UnboundVariable.java

```diff
@@ -21,40 +21,15 @@
 
 package com.vaticle.typeql.lang.pattern.variable;
 
-import com.vaticle.typeql.lang.pattern.constraint.ConceptConstraint;
-import com.vaticle.typeql.lang.pattern.constraint.Constraint;
-import com.vaticle.typeql.lang.pattern.constraint.ThingConstraint;
-import com.vaticle.typeql.lang.pattern.constraint.TypeConstraint;
-import com.vaticle.typeql.lang.pattern.variable.builder.ConceptVariableBuilder;
-import com.vaticle.typeql.lang.pattern.variable.builder.ThingVariableBuilder;
-import com.vaticle.typeql.lang.pattern.variable.builder.TypeVariableBuilder;
+import com.vaticle.typeql.lang.common.exception.TypeQLException;
 
-import java.util.Collections;
-import java.util.List;
-
-public class UnboundVariable extends Variable implements ConceptVariableBuilder,
-                                                         TypeVariableBuilder,
-                                                         ThingVariableBuilder.Common<ThingVariable.Thing>,
-                                                         ThingVariableBuilder.Thing,
-                                                         ThingVariableBuilder.Relation,
-                                                         ThingVariableBuilder.Attribute {
+import static com.vaticle.typedb.common.util.Objects.className;
+import static com.vaticle.typeql.lang.common.exception.ErrorMessage.INVALID_CASTING;
 
+public abstract class UnboundVariable extends Variable {
```

Comment:
Make abstract base class of UnboundDollarVariable and UnboundValueVariable.
Move implemented methods to those classes.